src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/McpServer/McpServer-CSharp/Program.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using ModelContextProtocol.Server;
+using System.ComponentModel;
+
+var builder = Host.CreateApplicationBuilder(args);
+
+if (args.Length == 0 || args[0] != "start-mcp")
+{
+    Console.Error.WriteLine("Error: invalid command. Use the 'start-mcp' command-line argument to start the MCP server.");
```

Comment:
Is this a pattern we want? why not just main starts it? I personally dislike this pattern I've seen in a few others (azure)

Reply:
UNderstood -- the article/quickstart can address those -- templates are durable and not only serving one purpose...so we should ensure any template we have be the 'correct' structure/we want and enable quickstarts/docs to cover some of the 'how'